File: src/jp/senchan/android/wasatter/WasatterItem.java
package jp.senchan.android.wasatter;

import java.io.Serializable;
import java.util.ArrayList;

import android.text.SpannableStringBuilder;

/**
 * Wasatterつぶやき＆ヒトコトクラス
 * 
 * @author Senka/Takuji
 * 
 */
public class WasatterItem implements Serializable {
    /**
	 *
	 */
    private static final long serialVersionUID = 1L;
    /**
     * ユーザーID（発言者）
     */
    public String id;
    /**
     * ユーザー名（発言者）
     */
    public String name;
    /**
     * ヒトコト
     */
    public String text;
    /**
     * プロフィール画像のURL
     */
    public String profileImageUrl;
    /**
     * パーマリンク
     */
    public String link;
    /**
     * RID
     */
    public String rid;
    /**
     * サービスの種類
     */
    public String service;
    public String replyUserNick;
    public String replyMessage;
    public long epoch;
    public ArrayList<String> favorite = new ArrayList<String>();
    public boolean channel;

    @Override
    public String toString() {
        // TODO 自動生成されたメソッド・スタブ
        return new SpannableStringBuilder(name).append("(").append(id)
                .append(")").toString();
    }

    public boolean favorited;
}

File: app/src/main/java/jp/senchan/android/wasatter/WasatterItem.kt
package jp.senchan.android.wasatter

import android.text.SpannableStringBuilder
import twitter4j.Status
import java.io.Serializable
import java.util.*

/**
 * Wasatterつぶやき＆ヒトコトクラス
 *
 * @author Senka/Takuji
 */
class WasatterItem(
        /**
         * ユーザーID（発言者）
         */
        val id: String,
        /**
         * ユーザー名（発言者）
         */
        val name: String?,
        /**
         * ヒトコト
         */
        val text: String?,
        /**
         * プロフィール画像のURL
         */
        val profileImageUrl: String,
        /**
         * パーマリンク
         */
        val link: String,
        /**
         * RID
         */
        val rid: String,
        /**
         * サービスの種類
         */
        val service: String, val replyUserNick: String?, val replyMessage: String?, val epoch: Long, val favorite: ArrayList<String?>?, val favorited: Boolean) : Serializable {

    constructor(status: Status) : this(
            status.user.screenName,
            status.user.screenName,
            status.text,
            status.user.profileImageURLHttps,
            "https://twitter.com/" + status.user.screenName + "/status/" + status.id, status.id.toString(),
            "Twitter",
            status.inReplyToScreenName,
            "",
            status.createdAt.time / 1000L,
            ArrayList<String?>(),
            status.isFavorited
    ) {
    }

    override fun toString(): String { // TODO 自動生成されたメソッド・スタブ
        return SpannableStringBuilder(name).append("(").append(id).append(
                ")").toString()
    }

    companion object {
        /**
         *
         */
        private const val serialVersionUID = 1L
    }

}